### PR TITLE
Support to set header names of `ClientRequest` as Camel-Case

### PR DESCRIPTION
**Pull request by PeterDing**
This is the case for supporting to request for servers which don't perfectly implement the `RFC 7230`. It is important for an app which uses `ClientRequest` as core part.

Because `http` crate will have no plans to support the Camel-Case headers, ([see](https://github.com/hyperium/http/issues/228#issuecomment-404996750)), We need to add an option to support it, as `hyper` [did](https://github.com/hyperium/http/issues/228#issuecomment-404996755).

So, I implement this case, as following.

* Add field `upper_camel_case_headers` to `ClientRequest`.

* Add function `set_upper_camel_case_headers` to `ClientRequest`
  and `ClientRequestBuilder` to set field `upper_camel_case_headers`.
  `true` is for turning on Camel-Case headers.
  `false` to disable it.
  Default is `false`.

* Add trait `client::writer::UpperCamelCaseHeader` for
  `http::header::HeaderName`, let it can be converted to Camel-Case
  then writed to buffer.

* Add test `test_client::test_upper_camel_case_headers`.

### CODE REVIEW

**Comment by fafhrd91** [comment]
Let’s remove upper, I think camel case is enough

**Comment by fafhrd91** [comment]
this PR should be moved to actix-http repo

**Comment by fafhrd91** [comment]
@PeterDing could you update PR for current master

File: actix-http/src/message.rs [hunk 1]
```diff
@@ -40,6 +40,7 @@ pub struct RequestHead {
     pub method: Method,
     pub version: Version,
     pub headers: HeaderMap,
+    pub upper_camel_case_headers: bool,
```
Comment (fafhrd91) [inline actix-http/src/message.rs]: Let's use Flags instead

**Comment by PeterDing** [comment]
@fafhrd91 I only add code to support the upper camel case for `RequestHead`.

**Comment by fafhrd91** [comment]
RequestHead is fine

**Comment by fafhrd91** [comment]
thanks!

**Pull request merged**